Code/GraphMol/Fingerprints/FingerprintGenerator.cpp

```diff
@@ -115,31 +111,36 @@ std::string FingerprintGenerator<OutputType>::infoString() const {
 }
 
 template <typename OutputType>
-SparseIntVect<OutputType>
-    *FingerprintGenerator<OutputType>::getFingerprintHelper(
-        const ROMol &mol, const std::vector<std::uint32_t> *fromAtoms,
-        const std::vector<std::uint32_t> *ignoreAtoms, const int confId,
-        const AdditionalOutput *additionalOutput,
-        const std::vector<std::uint32_t> *customAtomInvariants,
-        const std::vector<std::uint32_t> *customBondInvariants,
-        const std::uint64_t fpSize) const {
-  if (additionalOutput) {
-    if (additionalOutput->atomCounts) {
-      additionalOutput->atomCounts->resize(mol.getNumAtoms());
-      std::fill(additionalOutput->atomCounts->begin(),
-                additionalOutput->atomCounts->end(), 0);
+std::unique_ptr<SparseIntVect<OutputType>>
+FingerprintGenerator<OutputType>::getFingerprintHelper(
+    const ROMol &mol, FingerprintFuncArguments &args,
+    const std::uint64_t fpSize) const {
+  const ROMol *lmol = &mol;
+  std::unique_ptr<ROMol> tmol;
+  if (dp_fingerprintArguments->df_includeChirality &&
+      !mol.hasProp(common_properties::_StereochemDone)) {
+    tmol = std::unique_ptr<ROMol>(new ROMol(mol));
+    MolOps::assignStereochemistry(*tmol);
+    lmol = tmol.get();
+  }
+
+  if (args.additionalOutput) {
+    if (args.additionalOutput->atomCounts) {
+      args.additionalOutput->atomCounts->resize(lmol->getNumAtoms());
+      std::fill(args.additionalOutput->atomCounts->begin(),
+                args.additionalOutput->atomCounts->end(), 0);
```

Comment:
Perhaps clear atomCounts if we aren't using them in case args is being used more than once.  Otherwise document this behavior since this is a public api.

Reply:
Same with atomToBits and bitInfoMap.

Reply:
I'm not sure what you mean.... If the pointer is set for `atomCounts`, `atomToBits`, and/or `bitInfoMap` then they are being used. In this case they are already either resized and then filled with default values or they are cleared.

Reply:
I see that now, I misread the code.